### PR TITLE
fix deprecations

### PR DESCRIPTION
**Pull request by abhijithch**
Fixes ew deprecations on `Julia v0.5`.

### CODE REVIEW

File: src/tail.jl [hunk 1]
```diff
@@ -79,7 +79,7 @@ end
 
 # Trampolining
 
-trampname(f) = symbol(string("#__", f, "_tramp__"))
+trampname(f) = Symbol(string("#__", f, "_tramp__"))
```
Comment (tkelman) [inline src/tail.jl]: needs `@compat` for 0.3

Comment (abhijithch) [inline src/tail.jl]: Done.

File: src/Lazy.jl [hunk 2]
```diff
@@ -147,7 +147,7 @@ next(::List, xs::List) = first(xs), tail(xs)
 Base.show(io::IO, xs::List) =
   foreach(x->print(io,x), ["("] * interpose(xs, " ") * [")"])
 
-function Base.writemime(io::IO, ::MIME"text/plain", xs::List)
+@compat function show(io::IO, ::MIME"text/plain", xs::List)
```
Comment (tkelman) [inline src/Lazy.jl]: needs `using Compat`

Comment (abhijithch) [inline src/Lazy.jl]: Have pinned `Compat` and included `using Compat` at the very beginning. However tests are failing on all three `Julia` versions. But none seems to be related to deprecations.

Comment (tkelman) [inline src/Lazy.jl]: The package fails to even load in 0.3, so it should not claim to support it in REQUIRE if support can not be brought back there.

**Comment by MikeInnes** [comment]
Cool, thanks

**Comment by tkelman** [comment]
@MikeInnes please fix your tests at least on release julia, and don't claim to support releases in REQUIRE that the code doesn't work on.

**Pull request merged**